File: C++/OOP/Encapsulation/Encapsulation (example 3).cpp
#include <iostream>
using namespace std;

class Student{
  private:
    string name, dept;
    void academicInfo(string department) {
        cout << "Institute of Engineering" << endl; // just a random name
        cout << "Dept. of " << department << endl;
    }

  public:
    void setName(string studentName){
        name = studentName;
    }
    void setDept(string studentDept){
        dept = studentDept;
    }

    void displayInfo(){
        academicInfo(dept);
        cout << "Name: " << name << endl;
    }
};

int main(){
    Student naiem;
    naiem.setName("Abdullah Al Naiem");
    naiem.setDept("Computer");
    naiem.displayInfo();
    
    cout << endl;
    
    Student sakib;
    sakib.setName("Sakibur Rahman");
    sakib.setDept("Electrical");
    sakib.displayInfo();
}

File: Python/OOP/Encapsulation/Encapsulation (example 3).py
# Python does not have the private keyword, unlike some other object oriented languages.
# But encapsulation can be done with the following method (member can be private by double underscore)
class Student:
    def __academicInfo(self, department):
        print("Institute of Engineering"); # just a random name
        print("Dept. of " + department)

    def setName(self, studentName):
        self.__name = studentName

    def setDept(self, studentDept):
        self.__dept = studentDept

    def displayInfo(self):
        self.__academicInfo(self.__dept)
        print("Name: ", self.__name)


class encap:
    naiem = Student()
    naiem.setName("Abdullah Al Naiem")
    naiem.setDept("Computer")
    naiem.displayInfo()

    print()

    sakib = Student()
    sakib.setName("Sakibur Rahman")
    sakib.setDept("Electrical")
    sakib.displayInfo()